### PR TITLE
Dockerized test environment

### PR DESCRIPTION
**Pull request by jploskonka**
I found it quite tricky to figure out how to run all codecept tests. I would really love to have it possible to be able to run all tests with one command. Also manually running php server for helpers tests was little bit irritating for me.

I use Docker a lot in development and don't want to install Selenium locally. @DavertMik I would like to get your opinion on that if it's something you'd consider merging and suggesting as simpler way to contribute.

If you like the idea I can cleanup the code and finish it sometime soon. There's still issue with Appium tests (I didn't have time to fit it yet).

## Todo:
- [x] easy way to run single helper tests
- [x] Cleanup code
- [x] green travisCI and Semaphore
- [x] rest test
- [x] move docker-compose to test dir
- [x] acceptance test
- [x] support node 6 & node 8
- [x] Update travis config to use dockerized tests
- [x] README info about using it for development :)

### CODE REVIEW

File: test/helper/Nightmare_test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 let Nightmare = require('../../lib/helper/Nightmare');
 let should = require('chai').should();
 let I, browser;
-let site_url = 'http://127.0.0.1:8000';
+let site_url = 'http://php:8000';
```
Comment (DavertMik) [inline test/helper/Nightmare_test.js]:
```
(process.env.SITE_URL || 'http://127.0.0.1:8000');
```

here as well

**Comment by DavertMik** [comment]
Thanks @jploskonka . This is really great initiative. If you make it work with Appium I will adore you )
I couldn't make it run in Docker. It would be great to have headless mobile tests using Docker. If you make it work I'd add it to documentation as well... 

Also, I'd propose to move docker-compose.yml into tests or rename it to `docker-compose-test.yml` so people knew this file is for internal testing, not for starting a project itself. 

And there should be a flexible and documented way to run exact one helper and one test. As running all of them takes lot of time

**Comment by jploskonka** [comment]
I played a bit with Appium today and unfortunately wasn't able to run it :/ My container is always stuck on booting emulator so far:
novnc Not sure if it's stuck or just takes so long to boot on my machine :D 

I will give it one more shot this week, if I can't figure anything out I'll focus on getting rest of todos done ;) Still even without appium I think it's beneficial already and we can always work on Appium later. Thanks for your feedback!

**Comment by DavertMik** [comment]
> I will give it one more shot this week, if I can't figure anything out I'll focus on getting rest of todos done ;) Still even without appium I think it's beneficial already and we can always work on Appium later. Thanks for your feedback!

Yep, and thanks for your input 🤗. That would be really great to have easy to setup test environment

**Comment by jploskonka** [comment]
@DavertMik Another batch of updates, it's getting more interesting :D 

I have problems with `test/acceptance`. Specs for WebDriver passes OK, but nightmare fails on iframe related examples. I couldn't find info about those specs in `contributing.md` and also it looks like those aren't run on travisCI nor semaphore. Am I doing something wrong or are those outdated and maybe broken?
I'm also not 100% convinced to separate `docker-compose` services for each acceptance test, but it's easier to debug it for now ;)

**Comment by DavertMik** [comment]
> I have problems with test/acceptance. Specs for WebDriver passes OK, but nightmare fails on iframe related examples. I couldn't find info about those specs in contributing.md and also it looks like those aren't run on travisCI nor semaphore. Am I doing something wrong or are those outdated and maybe broken?

I didn't work with them for a long time so everything is possible 🙃 
Please proceed, some tests may be disabled. Esp for Nightmare...

> Also I noticed that acceptance specs are tagged with @protractor while there's no protractor codecept config ;)

I found it only here:

```js
Scenario('within on form @WebDriverIO @protractor @nightmare', (I) => {
```
In this case I think `@protractor` corresponds to SeleniumWebdriver helper

**Comment by jploskonka** [comment]
Almost done :D I still see issues mostly with travisCI integration but I'm not sure if this should be tackled in this PR or some other:
- Builds against node 6.9.5 are failing on travis, I didn't experienced it locally yet. I'd guess those are timeouts issues but I'm not sure.
- Because of building codecept container total build time on Travis is significantly longer
- I no longer test no-docker version of codecept which may lead to some compatibility issues
- Acceptance tests are not run on Travis, would be nice to tweak it (especially after #755 got merged and added some useful tests there)
See builds https://travis-ci.org/Codeception/CodeceptJS/builds/293626424 and https://travis-ci.org/Codeception/CodeceptJS/builds/293621806

When I started working on this PR I didn't intended to use dockerized tests on travis yet, I did it only to check out how it would work and then couldn't resist going further :D I rolled back `.travis.yml` to version from master and builds are green. I think the best way would be to merge this and I'd like to work further on mentioned issues in separate PRs.

**Comment by DavertMik** [comment]
Thanks!
Maybe we could setup another CI for testing docker container only?
There are still few free CIs so why not to use them. I'd pick Wercker CI http://www.wercker.com as it's already based on Docker container and could be easily set up for it.

**Comment by jploskonka** [comment]
I played a bit with Wercker and wasn't able to even start containers. I encountered some strange issues with privileges inside containers and gave up for now. I don't have much time this week to work on it ;)

Basically there's a lot of places we can improve this, but I'd like to add next things as separate PRs. Please let me know what do you think is must have to update here so we can get this merged and I can work on CI integration when I have some more time, probably next week :)

**Comment by DavertMik** [comment]
Thanks, let's switch to something new from Wercker

**Pull request merged**